drv/transceivers-server/src/main.rs

```diff
@@ -287,28 +285,23 @@ impl ServerImpl {
             return Err(FpgaError::CommsError);
         }
 
-        let mut out = [0u8; 2]; // [status, SFF8024Identifier]
+        let mut out = [0u8; 1]; // [SFF8024Identifier]
 
         // Wait for the I2C transaction to complete
-        loop {
-            self.transceivers
-                .get_i2c_status_and_read_buffer(port, &mut out)?;
-            if out[0] & Reg::QSFP::PORT0_STATUS::BUSY == 0 {
-                break;
-            }
-            userlib::hl::sleep_for(1);
-        }
+        let status = self
+            .transceivers
+            .get_i2c_status_and_read_buffer(port, &mut out)?;
 
-        if out[0] & Reg::QSFP::PORT0_STATUS::ERROR == 0 {
-            match out[1] {
+        if status.error == FpgaI2CFailure::NoError {
+            match out[0] {
                 0x1E => Ok(ManagementInterface::Cmis),
                 0x0D | 0x11 => Ok(ManagementInterface::Sff8636),
                 i => Ok(ManagementInterface::Unknown(i)),
             }
         } else {
             // TODO: how should we handle this?
             // Right now, we'll retry on the next pass through the loop.
```

Comment:
this comment is no longer accurate, right?

Reply:
It actually is. The loop being referred to here is actually the `update_thermal_loop` function which will call `get_transceiver_interface` repeatedly (https://github.com/oxidecomputer/hubris/blob/master/drv/transceivers-server/src/main.rs#L361)

Reply:
ah, i see, yeah, that makes sense.